Riot/Modules/Settings/SettingsViewController.m

```diff
@@ -1231,6 +1238,27 @@ - (void)refreshSettings
         [self editNewPhoneNumberTextField];
         keepNewPhoneNumberEditing = NO;
     }
+    
+    // Update notification access
+    [self refreshSystemNotificationSettings];
+}
+
+- (void)refreshSystemNotificationSettings
+{
+    __weak typeof(self) weakSelf = self;
```

Comment:
You can use the `MXWeakify`/`MXStrongifyAndReturnIfNilMXStrongifyAndReturnIfNil` combo here. Note that it provides some logging in case the object has been freed, which can be useful for debugging.

Reply:
Fixed!